src/main/java/org/vaadin/directory/HtmlHeaderController.java

```diff
@@ -55,6 +55,17 @@ public void doFilter(
             } else {
                 urlIdentifier = uri.substring(uri.indexOf(ROUTE_ADDON) + ROUTE_ADDON.length());
             }
+
+            // Check for special cases
+            if (urlIdentifier == null || urlIdentifier.isEmpty()) {
+                chain.doFilter(request, response);
+                return;
+            }
```

Comment:
This looks like a logical duplication with the initial approach to let the lookup of the addon fail (I assume that is the behavior of `service.getAddon()` method if the `urlIdentifier` is invalid). It would return `oc = null` and end in the else block that calls the next filter in the chain.

What about:
```
// Remove version number and anything trailing if present (e.g. "addonname/1.0.0")
if(urlIdentifier != null && !urlIdentifier.isEmpty()) {
    int slashIndex = urlIdentifier.indexOf('/');
    urlIdentifier = (slashIndex == -1) ? urlIdentifier : urlIdentifier.substring(0, slashIndex);
}
```
?

Reply:
That should work 👍